sassc.c

```diff
@@ -231,6 +231,28 @@ void invalid_usage(char* argv0) {
 
 }
 
+void push_include_substring(struct Sass_Options* options, char const* begin, char const* end, char* existing_buffer) {
+    if (begin == end) {
+        return;
+    }
+
+    size_t const count = end - begin;
+    char* buffer = 0;
+    if (count >= BUFSIZE) {
+        buffer = malloc(count + 1);
+    } else {
+        buffer = existing_buffer;
+    }
+
+    memcpy(buffer, begin, count);
+    buffer[count] = '\0';
+    sass_option_push_include_path(options, buffer);
+
+    if (buffer != existing_buffer) {
+        free(buffer);
```

Comment:
This is not required; libsass frees it on deleting the compilation context.

Reply:
I looked into `sass_option_push_include_path` implementation. It copies the contents of the string it was passed and then `options` object owns the copy. The source's ownershup is left to the caller, so the temporary buffer has to be deallocated.

Reply:
Ah, you are right; i forgot about `sass_copy_c_string`.